pkg/skaffold/initializer/init.go

```diff
@@ -335,20 +335,56 @@ func generateSkaffoldConfig(k Initializer, buildConfigPairs []builderImagePair)
 	return pipelineStr, nil
 }
 
-// TODO: make more flexible for non-docker builders
-func printAnalyzeJSON(out io.Writer, skipBuild bool, dockerfiles []InitBuilder, images []string) error {
-	if !skipBuild && len(dockerfiles) == 0 {
-		return errors.New("one or more valid Dockerfiles must be present to build images with skaffold; please provide at least one Dockerfile and try again or run `skaffold init --skip-build`")
+// printAnalyzeJSON takes the automatically resolved builder/image pairs, the unresolved images, and the unresolved builders, and generates
+// a JSON string containing builder config information,
+func printAnalyzeJSON(out io.Writer, skipBuild bool, pairs []builderImagePair, unresolvedBuilders []InitBuilder, unresolvedImages []string) error {
+	if !skipBuild && len(unresolvedBuilders) == 0 {
+		return errors.New("one or more valid Dockerfiles must be present to build images with skaffold; please provide at least one Dockerfile and try again, or run `skaffold init --skip-build`")
```

Comment:
Does this error message need to be more general since we're not just looking for Dockerfiles anymore?

Reply:
I left it as Dockerfiles for now since that's still the only thing that's being looked for, but I update the message in the Jib follow-up PR https://github.com/GoogleContainerTools/skaffold/pull/2276

Reply:
ah gotcha, thanks!